pkg/quarterdeck/db/tokens.go

```diff
@@ -0,0 +1,102 @@
+package db
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"errors"
+	"time"
+
+	"github.com/rotationalio/ensign/pkg/quarterdeck/keygen"
+	"github.com/vmihailenco/msgpack/v5"
+)
+
+const (
+	nonceLength = 64
+	keyLength   = 64
+)
+
+// NewVerificationToken creates a token struct from an email address that expires in 7
+// days.
+func NewVerificationToken(email string) *VerificationToken {
+	return &VerificationToken{
+		Email:     email,
+		ExpiresAt: time.Now().AddDate(0, 0, 7),
+		Nonce:     keygen.AlphaNumeric(nonceLength),
```

Comment:
I don't recommend restricting the nonce or the key to just alphanumeric characters; a byte can be any number between 0 and 255; but alphanumeric restricts you to 48-122 (at least for unicode) making the nonce and the secret less random for a fixed length of bytes.

The `keygen` package is designed for human consumption (apikeys, passwords, e.g. all printable characters). The reason we don't simply base64 encode a crypto random byte array in `keygen` is because we want to control the character set.